### PR TITLE
Restrict zero match clauses in match statement

### LINKED ISSUE #25296: Zero match-clauses should be handled
**Description:**
If match-stmt does not include match-clause(s) it should be a syntax error, because according to the spec there should be at least one or more match-clauses

**Steps to reproduce:**

```
function func(any v){
    match v {
    }
}

public function main() {
    func(2);
}
```

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$Subject

Fixes #25296

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal [hunk 1]
```diff
@@ -0,0 +1,4 @@
+function func(any v){
```
Comment (anupama-pathirage) [inline compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal:1]: Missing license header.

Comment (rdulmina) [reply compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal:17]: Added thanx.

Comment (SupunS) [reply compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal:17]: We haven't added licence headers to any of the parser test bal-sources. The reason for that was, these parser tests operates at a very granular level, and tests a specific grammar rule in each test-case.

Having the license header means, parser has to parse additional grammar rules as well - then we no longer have that granularity in the test..

Can we stick to the same?

Comment (rdulmina) [reply compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal:1]: reverted

Comment (anupama-pathirage) [reply compiler/ballerina-parser/src/test/resources/statements/match-stmt/match_stmt_source_16.bal:1]: +1. Thanks for the info @SupunS

File: compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties [hunk 2]
```diff
@@ -235,7 +235,8 @@ error.more.field.match.patterns.after.rest.field=more field match patterns after
 error.action.as.a.wait.expr=''wait'' cannot be used with actions
 error.invalid.usage.of.var='var' is not a valid type descriptor
 error.more.match.patterns.after.rest.match.pattern=more match patterns after rest match pattern
-error-match-pattern-not-allowed=match pattern not allowed
+error.match.pattern.not.allowed=match pattern not allowed
+error.at.least.one.match.clause.required=error at least one match clause required
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:239]: shall we reword the error message?
"match statement is required to have one or more match clauses"

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 3]
```diff
@@ -11802,7 +11802,19 @@ private STNode parseMatchStatement() {
...
+        // Parse match clauses list
+        List<STNode> matchClausesList = new ArrayList<>();
+        while (!isEndOfMatchClauses(peek().kind)) {
+            STNode clause = parseMatchClause();
+            matchClausesList.add(clause);
+        }
+        STNode matchClauses =  STNodeFactory.createNodeList(matchClausesList);
+        // At least one match clause required
+        if (isNodeListEmpty(matchClauses)) {
+            return SyntaxErrors.addDiagnostic(openBrace,
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:11814]: shouldn't be returning here

**Review by SupunS — CHANGES_REQUESTED** [review]

File: compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties (hunk 2 above)
Comment (rdulmina) [reply compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties]: updated

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java (hunk 3 above)
Comment (rdulmina) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java]: updated thanx

**Review by SupunS — APPROVED** [review]

**Pull request merged**